src/devices/Sn74hc595/Sn74hc595.cs

```diff
@@ -0,0 +1,361 @@
+        private int _srclk;
+        private int _rclk;
+
+        private int _count;
+
+        /// <summary>
+        /// Initialize a new Sn74hc595 device connected through GPIO (uses 3-5 pins)
+        /// </summary>
+        /// <param name="pinMapping">The pin mapping to use by the binding.</param>
+        /// <param name="gpioController">The GPIO Controller used for interrupt handling.</param>
+        /// <param name="shouldDispose">True (the default) if the GPIO controller shall be disposed when disposing this instance.</param>
+        /// <param name="count">Count of (daisy-chained) shift registers. Default/minimum is 1.</param>
+        public Sn74hc595(PinMapping pinMapping, GpioController gpioController = null,  bool shouldDispose = true, int count = 1)
+        {
+            pinMapping.Validate();
+            if (gpioController == null)
+            {
+                gpioController = new GpioController();
+            }
+
+            _controller = gpioController;
+            _shouldDispose = shouldDispose;
+            _pinMapping = pinMapping;
+            _data = _pinMapping.Data;
+            _srclk = _pinMapping.SrClk;
+            _rclk = _pinMapping.RClk;
+            _count = count;
+            SetupPins();
+        }
+
+        /// <summary>
+        /// Initialize a new Sn74hc595 device connected through SPI (uses 3 pins)
+        /// </summary>
+        /// <param name="spiDevice">SpiDevice used for serial communication.</param>
+        /// <param name="count">Count of (daisy-chained) shift registers. Default/minimum is 1.</param>
+        public Sn74hc595(SpiDevice spiDevice, int count = 1)
+        {
+            _spiDevice = spiDevice;
+            _count = count;
+        }
+
+        /// <summary>
+        /// Initialize a new Sn74hc595 device connected through both SPI and GPIO (SPI for writing data; GPIO for configuration; use 4-5 pins)
+        /// </summary>
+        /// <param name="spiDevice">SpiDevice used for serial communication.</param>
+        /// <param name="pinMapping">The pin mapping to use by the binding</param>
+        /// <param name="gpioController">The GPIO Controller used for interrupt handling</param>
+        /// <param name="shouldDispose">True (the default) if the GPIO controller shall be disposed when disposing this instance</param>
+        /// <param name="count">Count of (daisy-chained) shift registers. Default/minimum is 1.</param>
+        public Sn74hc595(SpiDevice spiDevice, PinMapping pinMapping, GpioController gpioController = null, bool shouldDispose = true, int count = 1)
+                  : this(pinMapping, gpioController, shouldDispose, count)
+        {
+            _spiDevice = spiDevice;
+        }
+
+        /// <summary>
+        /// Count of shift units.
+        /// The number of (daisy-chained) shift registers. Minimum is 1.
+        /// Not the count of registers on a single unit.
+        /// </summary>
+        public int Count => _count;
```

Comment:
Nit: Might have to do with the fact that I haven't worked with one of these before, but Count feels a bit vague to refer to number of devices daisy chained. Not sure of what could be a better name though, `DaisyChainLength` or something more descriptive?

Reply:
Exactly. I wasn't sure what to call that either. We should come up with a general term since it isn't specific to this chip. Maybe @terrajobst has an idea on that.

Reply:
`DeviceChainCount`?

Reply:
That’s headed in a good direction. Here are some options:

- ‘DeviceChainCount’
- ‘DeviceCount’
- ‘ChainedDeviceCount’

The only confusing thing is what the count is when not chained. It would be weird if the count jumped from 0 to 2. I like ‘DeviceCount’ best for that reason. Chaining seems to be more the mechanism that is used to increase the device count. Thoughts?

Reply:
Going to go with `DeviceCount`.